### PR TITLE
Implement initial tool shed API tests.

### PR DESCRIPTION
**Pull request by jmchilton**
Start implementing API tests for tool shed functionality we are using - so we know what we are doing won't break things or will break things in a way we have understandable tests for and so we can migrate the API to FastAPI, Pydantic, TRS, etc...

Already fixes some things. It seems legacy_* decorators have exception handling that doesn't play well with FastAPI adapters at all.

## How to test the changes?
(Select all options that apply)
- [X] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy_test/base/populators.py [hunk 1]
```diff
@@ -93,6 +93,7 @@
 from galaxy.util.resources import resource_string
 from . import api_asserts
 from .api import ApiTestInteractor
+from .api_util import random_name
```
Comment (nsoranzo) [inline lib/galaxy_test/base/populators.py:96]:
```suggestion
from .api_util import random_name as get_random_name
```

File: lib/galaxy_test/base/populators.py [hunk 2]
```diff
@@ -1195,13 +1196,7 @@ def reimport_history(self, history_id, history_name, wait_on_history_length, exp
...
         return imported_history_id
 
-    def get_random_name(self, prefix=None, suffix=None, len=10):
-        # stolen from navigates_galaxy.py
-        return "{}{}{}".format(
-            prefix or "",
-            "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(len)),
-            suffix or "",
-        )
+    get_random_name = random_name
```
Comment (nsoranzo) [inline lib/galaxy_test/base/populators.py]:
```suggestion
```

File: lib/tool_shed/webapp/api/users.py [hunk 3]
```diff
@@ -32,22 +33,19 @@ def create(self, trans, payload, **kwd):
...
-            message = f"email: {email}, username: {username} - {message}"
-            user_dict["message"] = message
-            user_dict["status"] = "error"
-        else:
-            # Create the user.
-            user = self.__create_user(trans, email, username, password)
-            user_dict = user.to_dict(view="element", value_mapper=self.__get_value_mapper(trans))
-            user_dict["message"] = f"User '{str(user.username)}' has been created."
-            user_dict["url"] = web.url_for(controller="users", action="show", id=trans.security.encode_id(user.id))
+            raise exceptions.RequestParameterInvalidException(message)
```
Comment (nsoranzo) [inline lib/tool_shed/webapp/api/users.py:42]: :heart:

File: lib/galaxy_test/base/populators.py (hunk 2 above)
Comment (jmchilton) [reply lib/galaxy_test/base/populators.py]: This needed to be a class method - wasn't consuming self, etc... I've reworked this.

**Review by bgruening — APPROVED** [review]
Thanks @jmchilton!

**Pull request merged**